Source/WebKit/UIProcess/API/haiku/PageClientImplHaiku.cpp

```diff
@@ -41,9 +43,7 @@ namespace WebKit
 
     std::unique_ptr<DrawingAreaProxy> PageClientImpl::createDrawingAreaProxy()
     {
-#if USE(COORDINATED_GRAPHICS)
```

Comment:
the goal here was to allow building in both cases: coordinated graphics enabled or disabled. Removing these guards should not be needed: either coordinated graphics is enabled, and this code should be enabled. Or, coordinated graphics is disabled, and this code should also be disabled.

This is needed because the webkit1 version of haikuwebkit always disables coordinated graphics (it doesn't work well otherwise). So we should try to keep things building in both modes. Even if this part of the source is webkit2 specific (it is in Source/WebKit), it may also be useful to have this in place if we change our mind again about enabling or disabling coordinated graphics.

Reply:
I've opted to use `#if USE(COORDINATED_GRAPHICS) || USE(TEXTURE_MAPPER)`, since `DrawingAreaProxyCoordinatedGraphics` compiles fine with only `USE_TEXTURE_MAPPER`. `DrawingAreaProxyCoordinatedGraphics` will give a build error if neither of these are enabled though.

...but, it seems every other example I found in the code uses only `#if USE(COORDINATED_GRAPHICS)`, so that might not be right. Let me see how well things compile with `USE_COORDINATED_GRAPHICS` on, and then I might drop the ` || USE(TEXTURE_MAPPER)`.

Reply:
Well, enabling `USE_COORDINATED_GRAPHICS` opens a whole can of worms. I think leaving it disabled is good enough for now. It fixes the crash and lets us move on to other problems.